sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -907,17 +907,21 @@ def _build_spec_by_group(
                 task_name_to_component_spec[
                     subgroup.name] = subgroup_component_spec
 
-                # TODO: handler importer spec.
-
-                subgroup_container_spec = builder.build_container_spec_for_task(
-                    task=subgroup)
-
                 executor_label = subgroup_component_spec.executor_label
 
                 if executor_label not in deployment_config.executors:
-                    deployment_config.executors[
-                        executor_label].container.CopyFrom(
-                            subgroup_container_spec)
+                    if hasattr(subgroup, 'container_spec'):
```

Comment:
Given that we document both `container_spec` and `importer_spec` as class attributes (in the docstring of PipelineTask), `hasattr` seems to be a bit contradictory. 
How about we initialize them to None. and check `if subgroup.container_spec is not None` here?